runtime/parachains/src/hrmp/benchmarking.rs

```diff
@@ -315,9 +316,18 @@ frame_benchmarking::benchmarks! {
 		let capacity = Configuration::<T>::config().hrmp_channel_max_capacity;
 		let message_size = Configuration::<T>::config().hrmp_channel_max_message_size;
 
-		// make sure this channel doesn't exist
+		// this will consume more weight if a channel _request_ already exists, because it will need
+		// to clear the request.
+		assert_ok!(Hrmp::<T>::hrmp_init_open_channel(
```

Comment:
The more correct solution would be to have this the variant with the open channel as worst case and then refund if there wasn't any worst case.

Reply:
But it's a root call, so there isn't an `AccoutId` to return some fee (weight) to. When the block is built, it wouldn't be able to use the weight refund to include more extrinsics, because the block is already built when this runs, right? As in, it doesn't know at block building time that execution would result in a refund.

Reply:
For sure we can refund the weight. Currently the scheduler runs in `on_initialize` and if there is weight that it didn't used it can give it back to include more transactions.